### DISCUSSION TITLE
[ghidra scripts] pcode offset mapping

### DISCUSSION THREAD

**Discussion by mksonzien** [NationalSecurityAgency/ghidra#3706] [Q&A]
Hello
I want to highlight only a specific address out of the whole function pcode.
Please inquire if it is possible to implement this.

def getAddress(offset):
return currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(offset)

addr = getAddress(0x00112ff1)
func_name = functionManager.getFunctionContaining(addr)
function = getGlobalFunctions(str(func_name))[0]
results = ifc.decompileFunction(function, 0, ConsoleTaskMonitor())
pcode = results.getDecompiledFunction().getC()
// highlight(pcode, offset)

**Comment by dragonmacher** [comment]
This is now possible in the master branch.  (Ghidra 10.2 will contain this new service.)   The following pseudo code shows how to get started when using a script:

```java
		PluginTool tool = getState().getTool();
		DecompilerHighlightService highlightService =
			tool.getService(DecompilerHighlightService.class);
		CTokenHighlightMatcher matcher = new CTokenHighlightMatcher() {

			@Override
			public Color getTokenHighlight(ClangToken token) {

				if (tokenIsInteresting(token)) {
					return Color.GREEN;
				}
				return null;
			}
		};

		DecompilerHighlighter highlighter =
			highlightService.createHighlighter(getClass().getSimpleName(), matcher);
		highlighter.applyHighlights();
```

**Reply by mksonzien** [reply]
I'm sorry for answering late.
Thank you so much for your reply.
I think it'll be a great help.

Hope you have a great weekend.
